File: ga/4omini_result_2024_7/result/1968C_mini/24_1968C.c
use std::io::{self, BufRead};

const N: usize = 500;
const A: usize = 1_000_000_000;

fn proc(x: &[usize], n: usize) -> Vec<usize> {
    let mut a = vec![0; n];
    a[0] = x[1] + 1;

    for i in 1..n {
        for j in 0..=A {
            a[i] = x[i] + j * a[i - 1];
            if (if i + 1 < n { x[i + 1] } else { 1 }) < a[i] && a[i] != a[i - 1] {
                break;
            }
        }
    }
    a
}

fn main() {
    let stdin = io::stdin();
    let reader = stdin.lock();
    let mut lines = reader.lines();
    
    let t: usize = lines.next().unwrap().unwrap().parse().unwrap();
    
    for _ in 0..t {
        let n: usize = lines.next().unwrap().unwrap().parse().unwrap();
        let mut x = vec![0; N];
        
        for i in 1..n {
            x[i] = lines.next().unwrap().unwrap().parse().unwrap();
        }
        
        let result = proc(&x, n);
        println!("{}", result.iter().map(|&v| v.to_string()).collect::<Vec<String>>().join(" "));
    }
}

File: ga/gpt3_5_result_2024_6/result/1968C_gpt/24_1968C.rs
use std::io::{self, BufRead};

const N: usize = 500;
const X: usize = 500;
const A: i32 = 1000000000;

fn proc() {
    let stdin = io::stdin();
    let mut handle = stdin.lock();
    
    let mut input = String::new();
    handle.read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    let mut x = vec![0; n];
    let mut a = vec![0; n];

    input.clear();
    handle.read_line(&mut input).unwrap();
    let mut iter = input.split_whitespace();

    for i in 1..n {
        x[i] = iter.next().unwrap().parse().unwrap();
    }

    a[0] = x[1] + 1;
    for i in 1..n {
        let mut j = 0;
        while j <= A {
            a[i] = x[i] + j * a[i - 1];
            if (i + 1 < n && x[i + 1] < a[i] || a[i] != a[i - 1]) {
                break;
            }
            j += 1;
        }
    }

    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!();
}

fn main() {
    let stdin = io::stdin();
    let mut handle = stdin.lock();

    let mut input = String::new();
    handle.read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        proc();
    }
}